lib/floe/workflow/error_matcher.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Floe
+  class Workflow
+    # Methods for common error handling
+    #
+    # @note Included class required to have `error_equals` available from the payload.
+    module ErrorMatcher
+      attr_reader :error_equals, :error_equals_all
+
+      def initialize(_workflow, name, payload)
+        @error_equals = payload["ErrorEquals"]
+        validate_error_equals!(name, ".#{payload["Type"]}")
+
+        @error_equals_all = error_equals.include?("States.ALL")
+        error_equals << "States.HeartbeatTimeout" if error_equals.include?("States.Timeout")
```

Comment:
I kept this optimization in a separate commit.

2 camps:
- modify this array and no special `if` cases.
- don't modify this array and add special `if` logic.